### PR TITLE
Add retry for metrics registration to fix ExternalCompactionMetricsIT

### PR DESCRIPTION
**Pull request by EdColeman**
Recently the ExternalCompactionMetricsIT has failed in Jenkins.  The logs seem to indicate that the it cannot find the metrics.

```
2024-01-31T06:20:26,199 [miniclusterImpl.MiniAccumuloClusterControl] DEBUG: Error getting running compactions from coordinator, message: Unable to get CompactionCoordinator address from ZooKeeper
```
If this does not resolve the issue, it will at least remove one potential cause. (I have not been able to reproduce this locally.)

This replaces #3983, rebased to 2.1.  If we decide on relocating Wait.waitFor() then the delay method added here can be replaced.

### CODE REVIEW

File: minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterControl.java [hunk 1]
```diff
@@ -153,17 +154,32 @@ public synchronized void startCoordinator(Class<? extends CompactionCoordinator>
...
-        }
+      long startNano = System.nanoTime();
+      if (!waitForMetrics()) {
+        log.warn("metrics failed to register after {} sec",
+            NANOSECONDS.toSeconds(System.nanoTime() - startNano));
+      }
+    }
+  }
+
+  private boolean waitForMetrics() {
```
Comment (dlmarion) [inline minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterControl.java:165]: Unless I'm missing something, it looks like this always returns `false`.

File: minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterControl.java [hunk 2]
```diff
@@ -153,17 +154,32 @@ public synchronized void startCoordinator(Class<? extends CompactionCoordinator>
       coordinatorProcess = cluster
           ._exec(coordinator, ServerType.COMPACTION_COORDINATOR, new HashMap<>()).getProcess();
       // Wait for coordinator to start
```
Comment (dlmarion) [inline minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterControl.java:156]: So this no longer waits for the coordinator to start, it only waits a certain amount of time. Is that right?

**Comment by EdColeman** [comment]
I'll rework this

**Comment by EdColeman** [comment]
This does not appear that it would correct the underlying issue that is causing the external compaction tests to fail.

**Pull request closed without merging**